frontends/PyCDE/src/pycde/value.py

```diff
@@ -258,6 +258,127 @@ def pad_or_truncate(self, num_bits: int):
   def __len__(self):
     return self.type.width
 
+  #  === Casting ===
+
+  def _exec_cast(self, targetValueType, type_getter, width: int = None):
+    from .dialects import hwarith
+    if type(self) is targetValueType:
+      return self
+    if width is None:
+      width = self.type.width
+    return hwarith.CastOp(self.value, type_getter(width))
+
+  def asInt(self, width: int = None):
+    """
+    Returns this value as a signless integer. If 'width' is provided, this value
+    will be truncated to that width.
+    """
+    return self._exec_cast(BitVectorValue, ir.IntegerType.get_signless, width)
+
+  def asSInt(self, width: int = None):
+    """
+    Returns this value as a a signed integer. If 'width' is provided, this value
+    will be truncated or sign-extended to that width.
+    """
+    return self._exec_cast(SignedBitVectorValue, ir.IntegerType.get_signed,
+                           width)
+
+  def asUInt(self, width: int = None):
+    """
+    Returns this value as an unsigned integer. If 'width' is provided, this value
+    will be truncated or zero-padded to that width.
+    """
+    return self._exec_cast(UnsignedBitVectorValue, ir.IntegerType.get_unsigned,
+                           width)
+
+  #  === Infix operators ===
+
+  # Signless operations. These will all return signless values - a user is
```

Comment:
Don't the operators need to be the same width? Does it make sense extend them automatically?

Reply:
> Don't the operators need to be the same width?
Correct - an error will be thrown during verification if this is the case. But, just as with the `hwarith` ops, i'll add an error so that a designer gets a warning on the python source line.

> Does it make sense extend them automatically?

If we are to keep the notion that PyCDE is a thing layer on top of CIRCT, then I would say no to this.